docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+  <https://www.sphinx-doc.org/en/master/usage/extensions/todo.html>`_ to support
+  ``todo`` directives
+* `nbsphinx <https://nbsphinx.readthedocs.io>`_ for including
+  `Jupyter`_ notebooks
+* `sphinx_copybutton <https://sphinx-copybutton.readthedocs.io>`_ to add
+  a "copy" button for code blocks
+* `sphinx_gallery.load_style
+  <https://sphinx-gallery.github.io/stable/advanced.html?highlight=load_style#using-only-sphinx-gallery-styles>`_
+  for using sphinx-gallery styles
+* IPython.sphinxext.ipython_console_highlighting
+* `sphinx_changelog <https://sphinx-changelog.readthedocs.io>`_
+  for rendering `towncrier`_ changelogs
+* `plasmapy_sphinx` for customizations created for use in PlasmaPy.
+  Note that `plasmapy_sphinx` is expected to be broken out into its own
+  package at a future date.
+
+These extensions are specified in ``extensions`` in `docs/conf.py`_.
+
+References to other packages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Intersphinx_ allows the automatic generation of links to the
+documentation of objects in other projects. The mappings are defined in
+the ``intersphinx_mapping`` dictionary in `docs/conf.py`_, and include
+`python`, `numpy`, `scipy`, `astropy`, `pandas`, `sphinx`, and
+`sphinx_automodapi`.
+
+When we include ``astropy.units.Quantity`` in the documentation, it
+will show up as `astropy.units.Quantity` with a link to Astropy's
+documentation.
+
+Substitutions
+~~~~~~~~~~~~~
+
+Some functions and classes are referred to repeatedly throughout the
+documentation. reST allows us to `define substitutions
+<https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#substitution-definitions>`_.
+
+.. code-block:: rst
+
+   .. |Particle| replace:: `~plasmapy.particles.particle_class.Particle`
+
+PlasmaPy has certain common substitutions pre-defined so that they can
+be used elsewhere in the documentation. For example, we can write
+``|Quantity|`` instead of ``~astropy.units.Quantity``, and
+``|Particle|`` instead of ``~plasmapy.particles.particle_class.Particle``.
+For an up-to-date list of substitutions, please refer to the
+`docs/common_links.rst`_ file.
+
+Because substitutions are performed when `sphinx` builds the
+documentation, they will not be performed before `help` accesses the
+docstring of an `object`. For example, when ``|Particle|`` is used in
+a docstring, `help` will show it as ``|Particle|`` rather than
+``~plasmapy.particles.particle_class.Particle``. Consequently,
+substitutions should not be used in docstrings when it is important
+that users have quick access to the full path of the `object` (such as
+in the ``See Also`` section).
+
+Writing documentation
+=====================
 
 Docstrings
-==========
+----------
+
+A docstring is a comment at the beginning of a function or another
+object that provides information on how to use that function.
+Docstrings begin with ``r"""`` (required when including backslashes,
+such as using LaTeX code in equations) or ``"""``, and end with
+``"""``.
```

Comment:
This would be a good spot to reference Python's PEP on docstrings, https://www.python.org/dev/peps/pep-0257/